tests/expecter.go

```diff
@@ -228,7 +229,9 @@ func ExpectBatchWithValidatedSend(expecter expect.Expecter, batch []expect.Batch
 
 			// Remove the \n since it is translated by the console to \r\n.
 			previousSend = strings.TrimSuffix(previousSend, "\n")
-			bExp.R = fmt.Sprintf("%s%s%s", previousSend, "((?s).*)", bExp.R)
+			bExp.R = fmt.Sprintf("%s%s%s|^\\s?%s*%s", previousSend, "((?s).*)", bExp.R, CRLF, bExp.R)
```

Comment:
Please use a raw string to avoid the `\\`.
As for the regex, I don't understand what this is doing. Maybe consider adding parentheses to make the operands of `|` more explicit.
Probably also deserves a comment.

Reply:
> Please use a raw string to avoid the `\\`.

Done.
> As for the regex, I don't understand what this is doing. Maybe consider adding parentheses to make the operands of `|` more explicit.

I added the parentheses, though I am not sure this is what you have meant, can you confirm?
> Probably also deserves a comment.

Done.

Reply:
I guess I was wondering about the precedence of `|`, but looks like it's low enough to be evaluated last.
What I meant for the parentheses was to make the left and right operands of `|` explicit, but it's literally `(everything on the left) | (everything on the right)`, so not really useful.
The comment helps a lot though, thank you!
@AlonaKaplan please have a quick look too.